### DISCUSSION TITLE
Why `?index` is pushed to the url when `<Form method="post">` is submitted?

### DISCUSSION THREAD

**Discussion by kicut** [remix-run/remix#7741] [Q&A]
When `<Form method="post">` is sent, `?index` is pushed to the URL, is this the expected behavior? Is there a way to stop this behavior?

To reproduce the problem open this link https://ll75wk-3000.csb.app , note the URL, press the Up/Down Vote button.

Codesanbox: https://codesandbox.io/p/sandbox/eloquent-ramanujan-ll75wk
Live: https://ll75wk-3000.csb.app/

https://github.com/remix-run/remix/assets/25601335/10c537ac-3691-4fae-9386-7cd4d65d3152

**Comment by kiliman** [comment]
In your example, the URL `/` matches both the *root* and *_index* routes. By default, Remix would POST to the parent route. Since only one `action` should be called (unlike loaders), Remix needs the `?index` to differentiate between the two.

The reason Remix defaults to the parent route is that you may start off with a route like `/users` (users.tsx). Then later add another route (users._index.tsx). Since they would both match the `/users` URL, it could cause problems if your app suddenly started calling a different action. By always calling the parent route, and explicitly calling the index route, you'll get consistent behavior.

It's documented here. 

https://remix.run/docs/en/main/components/form#index
